Tools/CollectSystemInfo.ps1

```diff
@@ -0,0 +1,197 @@
+        foreach ($path in $registryPaths) {
+            reg query $path | Out-File -FilePath virtio_disk.txt -Append
+        }
+        Write-Host "Virtio-win storage drivers registry info collection completed."
+    } catch {
+        Show-Warning "Failed to collect virtio-win storage drivers registry info: $_"
+    }
+}
+
+function Get-WindowsUpdateLogs {
+    Write-Host "Collecting Windows Update logs..."
+    try {
+        Get-Content -Path "$env:SystemRoot\WindowsUpdate.log" -ErrorAction SilentlyContinue | Out-File -FilePath WindowsUpdate.log
+        Write-Host "Windows Update logs collection completed."
+    } catch {
+        Show-Warning "Failed to collect Windows Update logs: $_"
+    }
+}
+
+function Get-WindowsUptime {
+    Write-Host "Collecting Windows uptime..."
+    try {
+        $uptime = (Get-Date) - (gcim Win32_OperatingSystem).LastBootUpTime
+        $uptime.ToString() | Out-File -FilePath WindowsUptime.txt
+        Write-Host "Windows uptime collection completed."
+    } catch {
+        Show-Warning "Failed to collect Windows uptime: $_"
+    }
+}
+
+function Get-ServicesList {
+    Write-Host "Collecting list of services..."
+    try {
+        $services = Get-Service
+        $services.ToString() | Out-File -FilePath Services.txt
+        Write-Host "Services collection completed."
+    }  catch {
+        Show-Warning "Failed to collect list of services: $_"
+    }
+}
+
+function Get-RunningProcesses {
+    Write-Host "Collecting list of running processes..."
+    try {
+        Get-Process | Select-Object -Property Id, ProcessName, StartTime | Export-Csv -Path RunningProcesses.csv -NoTypeInformation
+        Write-Host "Running processes collection completed."
+    } catch {
+        Show-Warning "Failed to collect list of running processes: $_"
+    }
+}
+
+function Get-InstalledApplications {
+    Write-Host "Collecting list of installed applications..."
+    try {
+        Get-ItemProperty -Path 'HKLM:\Software\Microsoft\Windows\CurrentVersion\Uninstall\*' |
+        Select-Object -Property DisplayName, DisplayVersion, Publisher, InstallDate |
+        Export-Csv -Path InstalledApplications.csv -NoTypeInformation
+        Write-Host "Installed applications collection completed."
+    } catch {
+        Show-Warning "Failed to collect list of installed applications: $_"
+    }
+}
+
+function Get-InstalledKBs {
+    Write-Host "Collecting list of installed KBs..."
+    try {
+        Get-HotFix | Select-Object -Property Description, HotFixID, InstalledOn | Export-Csv -Path InstalledKBs.csv -NoTypeInformation
+        Write-Host "Installed KBs collection completed."
+    } catch {
+        Show-Warning "Failed to collect list of installed KBs: $_"
+    }
+}
+
+function Get-WindowsMemoryDump {
+    Write-Host "Collecting Windows memory dump..."
+    if (Test-Path -Path "$env:SystemRoot\MEMORY.DMP") {
+        try {
+            Copy-Item -Path "$env:SystemRoot\MEMORY.DMP" -Destination ".\MEMORY.DMP" -Force

```

Comment:
MEMORY.DMP size can be up to RAM size, so compress it into ZIP instead of coping as is.